cli/command/load_balancer_test.go

```diff
@@ -0,0 +1,491 @@
+
+	contexts := map[string]*cli.Context{
+		"Missing NAME arg": getCLIContext(t, nil, nil),
+		"Missing PORT arg": getCLIContext(t, args{"name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := loadBalancerCommand.dropport(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+
+}
+
+func TestDisplayHealthCheck(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	loadBalancer := &models.LoadBalancer{
+		LoadBalancerID:   "lid",
+		LoadBalancerName: "lname",
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname"}, flags{})
+	if err := loadBalancerCommand.healthcheck(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUpdateHealthCheck(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	base.Resolver.EXPECT().
+		Resolve("load_balancer", "lname").
+		Return([]string{"lid"}, nil)
+
+	ports := []models.Port{}
+	healthCheck := models.HealthCheck{}
+
+	loadBalancer := &models.LoadBalancer{
+		HealthCheck:      healthCheck,
+		LoadBalancerID:   "lid",
+		LoadBalancerName: "lname",
+		Ports:            ports,
+	}
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(loadBalancer, nil)
+
+	healthCheck = models.HealthCheck{Target: "TCP:81"}
+	req := models.UpdateLoadBalancerRequest{
+		LoadBalancerID: "lid",
+		Ports:          &ports,
+		HealthCheck:    &healthCheck,
+	}
+
+	base.Client.EXPECT().
+		UpdateLoadBalancer(req).
+		Return("jid", nil)
+
+	job := &models.Job{
+		Status: "Completed",
+		Result: "lid",
+	}
+
+	base.Client.EXPECT().
+		ReadJob("jid").
+		Return(job, nil)
+
+	base.Client.EXPECT().
+		ReadLoadBalancer("lid").
+		Return(&models.LoadBalancer{}, nil)
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	c := getCLIContext(t, args{"lname"}, flags{"healthcheck-target": "TCP:81"})
+	if err := loadBalancerCommand.healthcheck(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestUpdateHealthCheck_UserInputError(t *testing.T) {
+	base, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+
+	contexts := map[string]*cli.Context{
+		"Non-int '--healthcheck-interval' flag": getCLIContext(t, args{"lname"}, flags{"healthcheck-interval": "two"}),
+		"Non-int '--healthcheck-timeout' flag":  getCLIContext(t, args{"lname"}, flags{"healthcheck-timeout": "two"}),
+		"Non-int '--healthy-threshold' flag":    getCLIContext(t, args{"lname"}, flags{"healthy-threshold": "two"}),
+		"Non-int '--unhealthy-threshold' flag":  getCLIContext(t, args{"lname"}, flags{"unhealthy-threshold": "two"}),
+		"Missing NAME arg":                      getCLIContext(t, nil, flags{"interval": 2}),
+	}
+
+	loadBalancerCommand := NewLoadBalancerCommand(base.Command())
+	for name, c := range contexts {
+		if err := loadBalancerCommand.healthcheck(c); err == nil {
```

Comment:
We're using the `t.Run` function to do table tests now. It gives us a bunch of cool benefits - we didn't know about it before. So this would be:
```go
for name, c := range contexts {
    t.Run(name, func(t *testing.T) {
        if err := loadBalancerCommand...
   })
}
```

Reply:
So here, I need to split this into two tests, because the missing name exits immediately, while the flag analysis happens after the load balancer entity is resolved and the load balancer is read.

Reply:
Actually, I figured out a way to test this in one test by adding some checking logic up front in the `healthcheck()` method